function/definition.go

```diff
@@ -85,6 +85,27 @@ func (d Definition) Parameter(ctx context.Context, position int) (Parameter, dia
 	return d.Parameters[position], nil
 }
 
+// ParameterName returns the Parameter name for a given argument position. This will be the return
+// from the `(Parameter).GetName` function or a default value. An error diagnostic is raised if the
+// position is outside the expected arguments.
+func (d Definition) ParameterName(ctx context.Context, position int) (string, diag.Diagnostics) {
```

Comment:
I put this logic here to allow the defaulting logic to be shared by the `internal/toproto{5/6}` packages and the `ValidateImplementation` logic below.

Perhaps there is another way to approach this?

Reply:
I'm good with the first suggestion here, the small amount of duplicated logic isn't a big deal (which is what I was originally trying to avoid)

Also updated the doc string for `(function.Parameter).GetName()`

Check out 858ddbb0662c5223614de73b952033018cb6f2ec